File: go-client-sdk/pkg/models/operations/getcompaniescompanyidreportsenhancedbalancesheetaccounts.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
	"time"
)

type GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsQueryParams struct {
	NumberOfPeriods *int       `queryParam:"style=form,explode=true,name=numberOfPeriods"`
	ReportDate      *time.Time `queryParam:"style=form,explode=true,name=reportDate"`
}

type GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsRequest struct {
	PathParams  GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsPathParams
	QueryParams GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsQueryParams
	Security    GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsSecurity
}

type GetCompaniesCompanyIDReportsEnhancedBalanceSheetAccountsResponse struct {
	CodatAssessDataContractsFinancialsStatementsEnhancedFinancialStatement *shared.CodatAssessDataContractsFinancialsStatementsEnhancedFinancialStatement
	ContentType                                                            string
	StatusCode                                                             int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidreportsenhancedbalancesheetaccounts.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}


export class GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=numberOfPeriods" })
  numberOfPeriods?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reportDate" })
  reportDate?: Date;
}


export class GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsSecurity;
}


export class GetCompaniesCompanyIdReportsEnhancedBalanceSheetAccountsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatAssessDataContractsFinancialsStatementsEnhancedFinancialStatement?: shared.CodatAssessDataContractsFinancialsStatementsEnhancedFinancialStatement;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
